infra.bs

```diff
@@ -1426,6 +1426,36 @@ out-of-bounds, except when used with <a for=list>exists</a>.
 "<code>b</code>", "<code>c</code>", "<code>a</code>" ». Then |example|[1] is the <a>string</a>
 "<code>b</code>".
 
+<p>For notational convenience, a multiple assignment syntax may be use to assign multiple variables
+to the <a>list</a>'s <a for=list>items</a>, by surrounding the variables to be assigned by « »
+characters and separating each variable name with a comma. The <a>list</a>'s <a for=list>size</a>
+must be the same as the number of variables to be assigned. Each variable given is then set to the
+value of the <a>list</a>'s <a for=list>item</a> at the corresponding index.
+
+<div class=example id=example-list-multiple-assignment>
+ <ol>
+  <li><p>Let |value| be the <a>list</a> « "<code>a</code>", "<code>b</code>", "<code>c</code>" ».
+
+  <li><p>Let « |a|, |b|, |c| » be |value|.
+
+  <li><p><a>Assert</a>: |a| is "<code>a</code>"
+
+  <li><p><a>Assert</a>: |b| is "<code>b</code>"
+
+  <li><p><a>Assert</a>: |c| is "<code>c</code>"
+ </ol>
+</div>
+
+<p class=note>It is recommended to check that the <a>list</a>'s <a for=list>size</a> is the
+expected length before using the list multiple assignment syntax.
+
+<div class=example id=example-list-multiple-assignment-size>
+ <ol>
+  <li><p><a>Assert</a>: <a>list</a>'s <a for=list>size</a> is <code>3</code>.
+  <li><p>Let « |a|, |b|, |c| » be <a>list</a>.
+ </ol>
+</div>
```

Comment:
> Should we also add an example where we have a size check before assignment?

Yeah, I think so. This is my first attempt at an example, but I'm not exactly sure if this is the best way to go about it is.

More contex form #509:

> I think I would prefer a two-step algorithm over handling a size mismatch inline, though I cannot really fault the English.

Does the added not and example sufficiently emphasize this?